packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -257,106 +196,178 @@ fn eval_file<'s>(scope: &mut v8::HandleScope<'s>, path: &str) -> Result<Value<'s
+            .clone();
+
+        let runner = match ModuleMap::import_module(
+            module_map,
             scope,
-            context,
-            "./src/worker/runner/javascript/hash_util.js",
-            &[arrow],
-        )?;
-        let batches_prototype = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/batch.js",
-            &[arrow, hash_util],
-        )?;
-
-        let ctx_import = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/context.js",
-            &[hash_util],
-        )?;
-        let ctx_import: Array<'_> = ctx_import.try_into().map_err(|err| {
-            Error::FileImport(
-                "./src/worker/runner/javascript/context.js".to_string(),
-                format!("Couldn't get array (of functions) from 'context.js': {err}"),
-            )
-        })?;
-        let experiment_ctx_prototype = ctx_import.get_index(scope, 0).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get experiment_ctx_prototype from index 0 of the context.js return \
-                 values"
-                    .to_string(),
-            )
-        })?;
-        let sim_init_ctx_prototype = ctx_import.get_index(scope, 1).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get sim_init_ctx_prototype from index 1 of the context.js return values"
-                    .to_string(),
-            )
-        })?;
-        let gen_ctx = ctx_import.get_index(scope, 2).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get gen_ctx from index 2 of the context.js return values".to_string(),
-            )
-        })?;
-
-        let gen_state = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/state.js",
-            &[hash_util],
-        )?;
-        let fns = import_file(
-            scope,
-            context,
             "./src/worker/runner/javascript/runner.js",
-            &[
-                arrow,
-                batches_prototype,
-                experiment_ctx_prototype,
-                sim_init_ctx_prototype,
-                gen_ctx,
-                gen_state,
-            ],
-        )?;
-        let fns: Array<'_> = fns.try_into().map_err(|err| {
-            Error::FileImport(
-                "./src/worker/runner/javascript/runner.js".into(),
-                format!("Couldn't get array (of functions) from 'runner.js': {err}"),
-            )
-        })?;
+        )? {
+            Some(runner) => runner,
+            None => {
+                return Err(Error::PackageImport(
+                    "./src/worker/runner/javascript/runner.js".to_string(),
+                    "Missing package".to_string(),
+                ));
+            }
+        };
+
+        // How to get a function from a module
+        // https://chromium.googlesource.com/v8/v8/+/refs/heads/lkgr/test/cctest/test-modules.cc#625
```

Comment:
It would be helpful if there are some comments on *our* code. The link could be helpful, too, but telling the reader some information like "functions are living in a namespace", and "read the function from the namespace" will help understand the code far more easily.

Reply:
Added some explanation in https://github.com/hashintel/hash/pull/493/commits/2ead12c83c7eb4c6dc00db1c4d38849b97fa635e